### PR TITLE
HID-2255: server-side validation of password reset, plus restore client-side validation

### PR DESCRIPTION
**Pull request by rupl**
# HID-2255

This started out as a small fix for the CSP, which disabled our client-side validation.

Since that happened, I'd actually noticed some intermittent errors during password resets and never could figure out what it was. It ends up that we weren't enforcing the `confirm_password` field on the server, which combined with lack of client-side validation, meant it was possible to change your password to something unexpected.

PR fixes both problems: we now have both server- and client-side validation, instead of nothing at all 🙈 

This also conveniently gets included in the 4.0.0 release, since the `confirm_password` property of the payload is now required.

## Testing

Basically, just reset a user's password and fill the form out incorrectly in various ways. It should now stop you from doing anything except the right thing.

To test server-side validation of form submissions, you can comment out the client-side JS file on your local, or disable JS in the browser.

### CODE REVIEW

**Review by left23 — APPROVED** [review]
@rupl Approving. I tested with multiple incorrect passwords, and with JS disabled, and this came up each time:
Selection_005
I also tested with the correct password criteria and I was able to reset.
